connectors/src/connectors/confluence/index.ts

```diff
@@ -0,0 +1,258 @@
+import {
+  ConnectorPermission,
+  ConnectorResource,
+  ModelId,
+} from "@dust-tt/types";
+import { ConnectorsAPIErrorResponse } from "@dust-tt/types";
+
+import { confluenceConfig } from "@connectors/connectors/confluence/lib/config";
+import {
+  getConfluenceCloudInformation,
+  listConfluenceSpaces,
+} from "@connectors/connectors/confluence/lib/confluence_api";
+import { ConfluenceSpaceType } from "@connectors/connectors/confluence/lib/confluence_client";
+import { ConnectorPermissionRetriever } from "@connectors/connectors/interface";
+import { Connector, sequelize_conn } from "@connectors/lib/models";
+import {
+  ConfluenceConfiguration,
+  ConfluenceSpace,
+} from "@connectors/lib/models/confluence";
+import { getAccessTokenFromNango } from "@connectors/lib/nango_helpers";
+import { Err, Ok, Result } from "@connectors/lib/result";
+import mainLogger from "@connectors/logger/logger";
+import { DataSourceConfig } from "@connectors/types/data_source_config";
+import { NangoConnectionId } from "@connectors/types/nango_connection_id";
+
+const { getRequiredNangoConfluenceConnectorId } = confluenceConfig;
+
+const logger = mainLogger.child({
+  connector: "confluence",
+});
+
+export async function createConfluenceConnector(
+  dataSourceConfig: DataSourceConfig,
+  connectionId: NangoConnectionId
+): Promise<Result<string, Error>> {
+  const nangoConnectionId = connectionId;
+  const confluenceAccessToken = await getAccessTokenFromNango({
+    connectionId: nangoConnectionId,
+    integrationId: getRequiredNangoConfluenceConnectorId(),
+    useCache: false,
+  });
+
+  const confluenceCloudInformation = await getConfluenceCloudInformation(
+    confluenceAccessToken
+  );
+  if (!confluenceCloudInformation) {
+    return new Err(new Error("Confluence access token is invalid"));
+  }
+
+  const { id: cloudId, url: cloudUrl } = confluenceCloudInformation;
+  try {
+    const connector = await sequelize_conn.transaction(async (transaction) => {
+      const connector = await Connector.create(
+        {
+          type: "confluence",
+          connectionId: nangoConnectionId,
+          workspaceAPIKey: dataSourceConfig.workspaceAPIKey,
+          workspaceId: dataSourceConfig.workspaceId,
+          dataSourceName: dataSourceConfig.dataSourceName,
+        },
+        { transaction }
+      );
+      await ConfluenceConfiguration.create(
+        {
+          cloudId,
+          connectorId: connector.id,
+          url: cloudUrl,
+        },
+        { transaction }
+      );
+
+      return connector;
+    });
+
+    // TODO(2024-01-10 flav) Uncomment in next PR.
+    // await launchConfluenceFullSyncWorkflow(connector.id, null);
+
+    return new Ok(connector.id.toString());
+  } catch (e) {
+    logger.error({ error: e }, "Error creating confluence connector.");
```

Comment:
Would be nice to add more info to the log Maybe creating a logger child with workspaceId (which is in datasourceconfig I thnk) / connectionId